oap-server/server-bootstrap/src/main/resources/lal/example.yaml

```diff
@@ -0,0 +1,43 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+rules:
+  - name: example
+    dsl: |
+      filter {
```

Comment:
I am feeling, we need a sampler before and after `text` to filter the logs, due to separate different services having different log formats.

Reply:
> I am feeling, we need a sampler before and after `text` to filter the logs, due to separate different services having different log formats.

Sampler is targeted on sink (persistent layer), it doesn't mean to serve as an `if` statement, users are free to have `if` statement in the parser


```groovy
filter {
  text {
    if log.service == "ServiceA" {
      text "log format of service A"
    } else if log.service == "ServiceB" {
      text "log format of service B"
    }
  }
}
```

Reply:
Make sense.